### ISSUE TITLE
Inconsistent File Format Handling Between Windows and POSIX in fast-glob

### ISSUE THREAD

**Issue by saoudi-h** [mrmlnc/fast-glob#425]
### Environment

  * OS Version: windows 11
  * Node.js Version: 20.7

### Actual behavior

Hello,

I'd like to report an issue that was initially raised on the Prettier repository https://github.com/prettier/prettier/issues/15426, but ultimately led us to examine Fast-glob. It turns out that Prettier uses Fast-glob to resolve patterns, especially for Prettier-cli. Specifically, in my case, it's about the following command: `npx prettier --write [...nextauth].ts`, which seems to be a filename format accepted on all operating systems.

It appeared to me that Fast-glob uses two different patterns, one for POSIX systems and another for Windows:

See: https://github.com/mrmlnc/fast-glob/blob/28a3d61e44d5d9193ba97de4f21df6dc7725f7c0/src/utils/path.ts#L13C1-L14C94

which ends up giving two different behaviors depending on whether you are on Linux or Windows.

I don't know the full implications of this, but I know that for my example the brackets "[]" in Windows are not escaped when we call the escapePath() method

This returns on Linux 
```pwsh
./myfolder/\[.myFile\].ts 
```

and on Windows. 
```pwsh
./myfolder/[.myfile].ts 
```

### Expected behavior

I believe that if Windows supports this naming format for these files, then Fast-glob should ensure it conforms to it.

### Steps to reproduce

on windows:
1. Create a small project:
```pwsh
mkdir fg-issue && cd fg-issue
npm init
npm i fast-glob
```

2. Create an index.js file with the following content:
```js
// index.js

const fg = require('fast-glob')

console.log(fg.escapePath('.[.any].any'))
```

4. execute :
pwsh or cmd: 
```pwsh
>node index

out on windows: .[.any].any

Out on linux : .\[.any\].any

```

### Code sample

A faster way to see the difference, although you'll have to trust me for the code I obtained from Fast-glob. 
I've included only the essentials to reproduce the problem:

```js

const path = "./any/[.any].ts";

const POSIX_UNESCAPED_GLOB_SYMBOLS_RE = /(\\?)([()*?[\]{|}]|^!|[!+@](?=\()|\\(?![!()*+?@[\]{|}]))/g;
const WINDOWS_UNESCAPED_GLOB_SYMBOLS_RE = /(\\?)([(){}]|^!|[!+@](?=\())/g;

function escapeWindowsPath(pattern) {
    return pattern.replace(WINDOWS_UNESCAPED_GLOB_SYMBOLS_RE, '\\$2');
}
function escapePosixPath(pattern) {
    return pattern.replace(POSIX_UNESCAPED_GLOB_SYMBOLS_RE, '\\$2');
}

console.log( "Posix : \n", escapePosixPath( path ) )
console.log( "Windows : \n", escapeWindowsPath( path ) )

```

**Comment by DamianGlowala** [comment]
This also affects my Nuxt project, where I use Prettier to format files on commit.

For instance, a file named `server/api/[...].ts` throws an error. This is a syntax Nuxt uses as a server-side [catch-all route](https://nuxt.com/docs/guide/directory-structure/server#catch-all-route).

**Comment by JetBoom** [comment]
Having this issue on an astro project

**Comment by kostia1st** [comment]
Using `husky` + `prettier` with Next.js routing patterns like `pages/home/[[...slugs]]/index.tsx` just got complicated. Still possible with `git commit -m "message" -n` though 😁

**Comment by mrmlnc** [comment]
Fixed in the `main` branch and was backported to `3.x.x`.

**Comment by saoudi-h** [comment]
The reported issue has been fixed. I'll go ahead and close this now. Thanks to mrmlnc for the quick resolution

**Comment by lapa182** [comment]
Was this released on npm under 3.3.2 or 3.3.1? I can't see in NPM :(.
